src/main/java/calculator/StringTokenParser.java

```diff
@@ -0,0 +1,38 @@
+package calculator;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class StringTokenParser implements TokenParser {
+
+    @Override
+    public TokenNumbers parseTokenNumbers(String input) {
+        String tokens = input;
+        String splitter = ",|:";
+
+        Pattern pattern = Pattern.compile("//(.)\\n(.*)");
```

Comment:
"//(.)\\n(.*)"와 같은 문자열은 상수로 정의해두는게 어떨까요?

Reply:
Pattern 객체의 경우 아래와 같이 상수로 정의해서 사용하는 건 어떨까요?
```suggestion
    private static final Pattern STRING_TOKEN_PATTERN = Pattern.compile("//(.)\\n(.*)");
```

지금처럼 메소드 내 지역 변수로 들어가게 된다면 parseTokenNumbers가 호출되는 시점마다 Pattern 객체가 생성될 것 같아요

관련 내용을 아래에 첨부해두겠습니다.

> 만드는데 메모리나 시간이 오래 걸리는 객체 즉 "비싼 객체"를 반복적으로 만들어야 한다면 캐시해두고 재사용할 수 있는지 고려하는 것이 좋다.

https://github.com/keesun/study/blob/master/effective-java/item6.md